appmonitoring/ts/src/CertificateGenerator.ts

```diff
@@ -0,0 +1,168 @@
+        caCert.setSubject(attributes);
+        caCert.setIssuer(attributes);
+    
+        const extensions = [{
+            name: 'basicConstraints',
+            cA: true
+        },
+        {
+            name: 'subjectKeyIdentifier',
+            keyIdentifier: caCert.generateSubjectKeyIdentifier().getBytes(),
+        },
+        {
+            name: 'keyUsage',
+            keyCertSign: true,
+            cRLSign: true,
+            digitalSignature: true,
+            keyEncipherment: true,
+        }];
+
+        caCert.setExtensions(extensions);
+        caCert.sign(caCert.privateKey,forge.md.sha256.create());
+
+        const caCertResult: CACertData = {
+            certificate: forge.pki.certificateToPem(caCert),
+            serviceKey: forge.pki.privateKeyToPem(caCert.privateKey)
+        }
+
+        const host_attributes = [{
+            shortName: 'CN',
+            value: "app-monitoring-webhook-service.kube-system.svc"
+        }];
+    
+        const host_extensions = [{
+            name: 'basicConstraints',
+            cA: false
+        }, 
+        {
+            name: 'authorityKeyIdentifier',
+            keyIdentifier: caCert.generateSubjectKeyIdentifier().getBytes(),
+        }, 
+        {
+            name: 'keyUsage',
+            digitalSignature: true,
+            keyEncipherment: true
+        },
+        {
+            name: 'extKeyUsage',
+            serverAuth: true
+        }, 
+        {
+            name: 'subjectAltName',
+            altNames: [{ type: 2, value: "app-monitoring-webhook-service.kube-system.svc" }]
+        }];
+
+        const newHostCert = forge.pki.createCertificate();
+        const hostKeys = forge.pki.rsa.generateKeyPair(4096);
+
+        // Set the attributes for the new Host Certificate
+        newHostCert.publicKey = hostKeys.publicKey;
+        newHostCert.serialNumber = CertificateManager.randomHexSerialNumber();
+        newHostCert.validity.notBefore = new Date(2023,6,20,0,0,0,0);
+        newHostCert.validity.notAfter = new Date(2025,7,21,0,0,0,0);
+        newHostCert.setSubject(host_attributes);
+        newHostCert.setIssuer(caCert.subject.attributes);
+        newHostCert.setExtensions(host_extensions);
+
+        // Sign the new Host Certificate using the CA
+        newHostCert.sign(caCert.privateKey, forge.md.sha256.create());
+
+        // // Convert to PEM format
+        const pemHostCert = forge.pki.certificateToPem(newHostCert);
+        const pemHostKey = forge.pki.privateKeyToPem(hostKeys.privateKey);
+
+        return {
+            caCert: caCertResult.certificate,
+            tlsCert: pemHostCert,
+            tlsKey: pemHostKey
+        } as WebhookCertData;
+    }
+
+    public static async PatchSecretStore(kubeConfig: k8s.KubeConfig, certificate: WebhookCertData) {
+        const secretsApi = kubeConfig.makeApiClient(k8s.CoreV1Api);
+        const secretStore = await secretsApi.readNamespacedSecret("app-monitoring-webhook-cert", "kube-system");
```

Comment:
Any way to not have the name of the secret in two places? Same as with service FQND.